src/main/kotlin/no/nav/klage/oppgave/service/PersonsoekService.kt

```diff
@@ -0,0 +1,78 @@
+package no.nav.klage.oppgave.service
+
+import no.nav.klage.oppgave.api.mapper.KlagebehandlingerSearchCriteriaMapper
+import no.nav.klage.oppgave.api.view.PersonSoekInput
+import no.nav.klage.oppgave.clients.pdl.graphql.PdlClient
+import no.nav.klage.oppgave.clients.pdl.graphql.SoekPersonResponse
+import no.nav.klage.oppgave.domain.elasticsearch.EsKlagebehandling
+import no.nav.klage.oppgave.domain.personsoek.PersonSoekResponse
+import no.nav.klage.oppgave.domain.personsoek.PersonSoekResponseList
+import no.nav.klage.oppgave.util.getLogger
+import no.nav.klage.oppgave.util.getSecureLogger
+import org.springframework.stereotype.Service
+import java.lang.RuntimeException
+import java.time.LocalDate
+
+@Service
+class PersonsoekService(
+    private val pdlClient: PdlClient,
+    private val elasticsearchService: ElasticsearchService,
+    private val klagebehandlingerSearchCriteriaMapper: KlagebehandlingerSearchCriteriaMapper
+) {
+    companion object {
+        @Suppress("JAVA_CLASS_ON_COMPANION")
+        private val logger = getLogger(javaClass.enclosingClass)
+        private val secureLogger = getSecureLogger()
+    }
+
+    fun personsoek(navIdent: String, input: PersonSoekInput): PersonSoekResponseList {
+        return if(input.isFnrSoek()) {
+            fnrSoek(navIdent, input)
+        } else {
+            navnSoek(navIdent, input)
+        }
+    }
+
+    private fun fnrSoek(navIdent: String, input: PersonSoekInput): PersonSoekResponseList {
+        val liste = esSoek(navIdent, input)
+        val mapped = liste.groupBy { it.sakenGjelderFnr }.map { (key, value) ->
+            PersonSoekResponse(
+                fnr = key!!,
+                navn = value.first().sakenGjelderNavn,
+                foedselsdato = null,
+                klagebehandlinger = value
+            )
+        }
+        return PersonSoekResponseList(liste.size, mapped)
+    }
+
+    private fun navnSoek(navIdent: String, input: PersonSoekInput): PersonSoekResponseList {
+        val pdlResponse = pdlClient.personsok(input.soekString)
+        verifyPdlResponse(pdlResponse)
+        val mapped = pdlResponse.data?.soekPerson?.hits?.map {
+            val fnr = it.person.folkeregisteridentifikator.identifikasjonsnummer
+            val klagebehandlinger = esSoek(navIdent, input.copy(soekString = fnr))
```

Comment:
Her gjør du et søk mot ES per fnr du får fra PDL. Jeg la opp til at man kan legge inn en liste av fnr og gjøre ett søk mot ES. Ville det vært fordelaktig å bruke her, eller er det faktisk bedre for deg å gjøre flere søk? (Mindre opprydding i etterkant for å gruppere på person?)